x/ibc-hooks/wasm_hook.go

```diff
@@ -117,7 +111,21 @@ func (h WasmHooks) OnRecvPacketOverride(im IBCMiddleware, ctx sdk.Context, packe
 		return osmoutils.NewEmitErrorAcknowledgement(ctx, types.ErrWasmError, err.Error())
 	}
 
-	fullAck := ContractAck{ContractResult: response.Data, IbcAck: ack.Acknowledgement()}
+	// Check if the contract is requesting for the ack to be async.
+	var asyncAckRequest types.OnRecvPacketAsyncAckResponse
+	err = json.Unmarshal(response.Data, &asyncAckRequest)
+	if err == nil {
+		// If unmarshalling succeeds, the contract is requesting for the ack to be async.
+		if asyncAckRequest.IsAsyncAck { // in which case IsAsyncAck is expected to be set to true
+			// TODO: Do we want to check that only whitelisted contracts can do AsyncAcks?
```

Comment:
@ValarDragon also here. Any thoughts?

Reply:
Asking to learn: why do we want to whitelist this functionality?

Reply:
Discussed offline, whitelist makes sense to me